src/main/java/org/carlmontrobotics/subsystems/Arm.java

```diff
@@ -1,48 +1,209 @@
+        armMotor2.setIdleMode(IdleMode.kBrake);
+        armEncoder.setPositionConversionFactor(rotationToRad);
+        armEncoder.setVelocityConversionFactor(rotationToRad);
+        armEncoder.setInverted(encoderInverted);
+
+        armMotor2.follow(armMotor1);
+     
+        //armEncoder1.setZeroOffset(offsetRad);
+      
+        //armPID.setTolerance(posToleranceRad, velToleranceRadPSec);
+
+        SmartDashboard.putData("Arm", this);
+
+        //armProfileTimer.start(); <-- don't neeed timer anymore
+
+        setArmTarget(goalState.position);
+
+        // SmartDashboard.putNumber("Arm Max Vel", MAX_FF_VEL );
+        SmartDashboard.putNumber("ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD", ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD);
+        SmartDashboard.putNumber("Arm Tolerance Pos", posToleranceRad);
+        SmartDashboard.putNumber("Arm Tolerance Vel", velToleranceRadPSec);
+    }
+
+    @Override
+    public void periodic() {
+
+        if(DriverStation.isDisabled()) resetGoal();
+
+        //ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD = SmartDashboard.getNumber("ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD", ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD);
+        // armConstraints = new TrapezoidProfile.Constraints(MAX_FF_VEL , MAX_FF_ACCEL );
+        armPID1.setP(kP);
+        armPID1.setI(kI);
+        armPID1.setD(kD);
+        armPID2.setP(kP);
+        armPID2.setI(kI);
+        armPID2.setD(kD);
+
+        //smart dahsboard stuff
+        //SmartDashboard.putBoolean("ArmPIDAtSetpoint", armPID1.atSetpoint());
+        SmartDashboard.putBoolean("ArmProfileFinished", armProfile.isFinished(armProfileTimer.get()));
+        //posToleranceRad = SmartDashboard.getNumber("Arm Tolerance Pos", posToleranceRad);
+        //velToleranceRadPSec= SmartDashboard.getNumber("Arm Tolerance Vel", velToleranceRadPSec);
+
+       // SmartDashboard.putNumber("MaxHoldingTorque", maxHoldingTorqueNM());
+        //SmartDashboard.putNumber("V_PER_NM", getV_PER_NM());
+        SmartDashboard.putNumber("COMDistance", getCoM().getNorm());
+        SmartDashboard.putNumber("InternalArmVelocity", armEncoder.getVelocity());
+        //SmartDashboard.putNumber("Arm Current", armMotor.getOutputCurrent());
+
+       // SmartDashboard.putNumber("ArmPos", getArmPos());
+
+        //driveArm(armProfile.calculate(armProfileTimer.get()));
+
+        autoCancelArmCommand();
+    }
+
+  public void autoCancelArmCommand() {
+        if(!(getDefaultCommand() instanceof ArmTeleop) || DriverStation.isAutonomous()) return;
+
+        double requestedSpeeds = ((ArmTeleop) getDefaultCommand()).getRequestedSpeeds();
+
+        if(requestedSpeeds != 0) {
+            Command currentArmCommand = getCurrentCommand();
+            if(currentArmCommand != getDefaultCommand() && currentArmCommand != null) {
+                currentArmCommand.cancel();
+            }
+        }
+    }
+
+    //#region Drive Methods
+    public void driveArm(double goalAngle){
```

Comment:
This shouldn't be used outside of subsystem since it is constantly being run in `periodic()`. make it private